minerl/env/core.py

```diff
@@ -668,8 +668,19 @@ def _init_mission(self):
                 num_retries += 1
                 if num_retries > MAX_WAIT:
                     raise socket.timeout()
-                self.log_error("Did not get an OK from Malmo; trying again.")
-                time.sleep(1)
+                elif self.log_shows_bind_exception():
+                    # Skips error handler to directly to abort.
+                    self.had_to_clean = True
+                    logger.error("Malmo server failed to bind to port, possibly due "
+                                 "to collision with parallel Malmo instance. "
+                                 "Giving up on contacting this Malmo server, and "
+                                 "starting a new one. It's possible that the abandoned Malmo "
+                                 "server cannot be closed automatically, and you will "
+                                 "have to do so manually later.")
```

Comment:
Has this happened? It seems like we could try to kill it from python, at least using some processing or os API

Reply:
InstanceManager has a pretty extensive procedure for attempting to shutdown the Minecraft instance, starting from sending a "please shutdown gracefully" message to the assigned port and ending with a "OS says you must shutdown immediately via signal 9" at the (Minecraft?) PID.

I think that I often run into zombie Minecraft instances anyways even without port collisions when I KeyboardInterrupted my RL training (parallelized, ~12 instances). I'm not sure why I had to manually close stuff, but I ended up making an alias because I had to do it so often:

`alias cull_minecraft='sudo pkill -9 -f Minecraft'`

Reply:
I don't understand why we have zombie instances enough to fix this problem right now, but might revisit this in a later PR if this keeps on showing up.